src/extensions/replay/sessionrecording.ts

```diff
@@ -423,15 +423,15 @@ export class SessionRecording {
         const sessionIdChanged = this.sessionId !== sessionId
         const windowIdChanged = this.windowId !== windowId
 
+        this.windowId = windowId
+        this.sessionId = sessionId
+
         if (
             [FULL_SNAPSHOT_EVENT_TYPE, META_EVENT_TYPE].indexOf(event.type) === -1 &&
             (windowIdChanged || sessionIdChanged)
         ) {
             this._tryTakeFullSnapshot()
         }
-
```

Comment:
I know nearly nothing about this code base but by just looking at it, if the full snapshot isn't taken (hence missing) because the `if` condition returns false, I'm not sure the fix actually will solve the issue since the `windowIdChanged` and `sessionIdChanged` used the values before assignment anyway.

`_tryTakeFullSnapshot` also does not use the new `windowId` and `sessionId` values either.

If the snapshot is missing because of L430, I'd bet on the `[FULL_SNAPSHOT_EVENT_TYPE, META_EVENT_TYPE].indexOf(event.type) === -1` condition.

Feel free to ignore me, it's just what I can grasp by looking at the code diff, happy to get it merged and try it out.

Reply:
it's good to start getting you familiar with it!

it may well not be the only fix...

I'm assuming everything is synchronous... in which case calling tryTakeFullSnapshot (if it succeeds) will end up in `_captureSnapshotBuffered` the existing buffer will have session id A and this change will mean its called with session id b in the properties arg. 

In which case we'll flush the existing buffer, create a new buffer, and then queue the full snapshot for sending to the backend next time we do flush

My worry is with the current code we can get in a state where we add the full snapshot to the previous buffer, and then do th switcheroo with the next snapshot emitted

Reply:
Oh I see, so once the snapshot is taken, the `onRRwebEmit` callback is called (also `_captureSnapshotBuffered` internally) and then it uses the IDs, in this case, seems like a good try.